lib/services/types.proto

```diff
@@ -750,6 +750,14 @@ message RoleConditions {
 
     // KubeUsers is an optional kubernetes users to impersonate
     repeated string KubeUsers = 7 [ (gogoproto.jsontag) = "kubernetes_users,omitempty" ];
+
+    // ClusterLabels is a map of node labels (used to dynamically grant access to
+    // clusters).
+    wrappers.LabelValues ClusterLabels = 8 [
+        (gogoproto.nullable) = false,
+        (gogoproto.jsontag) = "cluster_labels,omitempty",
+        (gogoproto.customtype) = "Labels"
+    ];
```

Comment:
I wonder of we should call this something like `remote_cluster_labels` or `leaf_cluster_labels` to better clarify that these labels don't affect the user's access to the cluster in which the role is defined.  Its obvious that this is true if you understand teleport's internals, but may not be obvious to all users.

Reply:
I was thinking to extend the cluster labels feature to main cluster as well, we just lack controls over the cluster labels at the moment, but once implemented will complete the cycle :)